x-pack/plugins/xpack_main/server/lib/populate_ui_capabilities.ts

```diff
@@ -8,6 +8,8 @@ import _ from 'lodash';
 import { UICapabilities } from 'ui/capabilities';
 import { Feature } from '../../types';
 
+const ELIGIBLE_FLAT_MERGE_KEYS = ['catalogue'];
```

Comment:
Apologies for being dense and forgetful, where are we doing this similar functionality for "nav links"?

Reply:
One of our earlier implementations was using this for `navLinks`, but we've since moved that into its own [dedicated handler](https://github.com/legrego/kibana/blob/gap/catalogue/src/ui/ui_render/ui_render_mixin.js#L38-L49).

This is most similar to what we're doing for [merging the overall `uiCapabilities` injected variable](https://github.com/legrego/kibana/blob/gap/catalogue/src/ui/ui_render/lib/merge_variables.ts), rather than the feature properties within it.